src/HotChocolate/Language/src/Language.SyntaxTree/ObjectFieldNode.cs

```diff
@@ -125,17 +124,10 @@ public override bool Equals(object? obj)
     /// </returns>
     public override int GetHashCode()
     {
-        unchecked
-        {
-            if (_hash == null)
-            {
-                _hash = (Kind.GetHashCode() * 397)
-                    ^ (Name.GetHashCode() * 97)
-                    ^ (Value.GetHashCode() * 7);
-            }
-
-            return _hash.Value;
-        }
+        var hashCode = new HashCode();
```

Comment:
Since we are not dealing with lists or raw memory you can use `HashCode.Combine`. This will save us from creating a HashCode instance.

Further, we need to include the kind in the hash. We do not need it in the equals since we have the type but for the hashcode, we want to include it to have a better hashcode quality.

`HashCode.Combine(Kind, Name, Value)`